Create a new database called 'crudeoilprice'

Connect to the 'master' database to run this snippet

In [2]:
USE master

Commands completed successfully.

Total execution time: 00:00:00.001

Create the new database if it does not exist already

In [1]:
IF NOT EXISTS (
    SELECT [name]
        FROM sys.databases
        WHERE [name] = N'crudeoilprice'
)
CREATE DATABASE crudeoilprice

Commands completed successfully.

Total execution time: 00:00:00.034

In [3]:
SELECT *
FROM crudeoilprice.dbo.CrudeOilProdPrice

-- Delete 2021 rows from table '[CrudeOilProdPrice]' in schema '[dbo]'
DELETE FROM crudeoilprice.dbo.CrudeOilProdPrice
WHERE [Year] = 2021

(180 rows affected)

(0 rows affected)

Total execution time: 00:00:00.032

Year Month Crude_Oil_Price Production Crude_Oil_Export 2006 January 63.849998474121094 2.5899999141693115 2.140000104904175 2006 February 61.33000183105469 2.4700000286102295 2.0199999809265137 2006 March 65 2.25 1.7999999523162842 2006 April 72.08999633789062 2.319999933242798 1.8700000047683716 2006 May 71.18000030517578 2.2799999713897705 1.8300000429153442 2006 June 69.31999969482422 2.4100000858306885 1.9600000381469727 2006 July 75.12999725341797 2.390000104904175 1.940000057220459 2006 August 75.1500015258789 2.4000000953674316 1.9500000476837158 2006 September 62.970001220703125 2.4000000953674316 1.9500000476837158 2006 October 59.4900016784668 2.380000114440918 1.9299999475479126 2006 November 59.810001373291016 2.3399999141693115 1.8899999856948853 2006 December 64.69999694824219 2.359999895095825 1.909999966621399 2007 January 55.56999969482422 2.319999933242798 1.8700000047683716 2007 February 59.970001220703125 2.309999942779541 1.8600000143051147 2007 March 64.27999877929688 2.140000104904175 1.690000057220459 2007 April 70.45999908447266 2.1600000858306885 1.7100000381469727 2007 May 70.4000015258789 1.9600000381469727 1.5099999904632568 2007 June 73.27999877929688 2.0799999237060547 1.6299999952316284 2007 July 79.76000213623047 2.1700000762939453 1.7200000286102295 2007 August 73.76000213623047 2.200000047683716 1.75 2007 September 79.76000213623047 2.309999942779541 1.8600000143051147 2007 October 83.86000061035156 2.319999933242798 1.8700000047683716 2007 November 95.05000305175781 2.25 1.7999999523162842 2007 December 93.4000015258789 2.200000047683716 1.75 2008 January 94.26000213623047 2.1700000762939453 1.7200000286102295 2008 February 98.1500015258789 2.0799999237060547 1.6299999952316284 2008 March 103.7300033569336 2.059999942779541 1.6100000143051147 2008 April 116.7300033569336 1.9600000381469727 1.5099999904632568 2008 May 126.56999969482422 2.049999952316284 1.600000023841858 2008 June 138.74000549316406 2.0199999809265137 1.5700000524520874 2008 July 137.74000549316406 2.130000114440918 1.6799999475479126 2008 August 115.83999633789062 2.109999895095825 1.659999966621399 2008 September 103.81999969482422 2.1700000762939453 1.7200000286102295 2008 October 75.30999755859375 2.259999990463257 1.809999942779541 2008 November 55.5099983215332 2.140000104904175 1.690000057220459 2008 December 45.869998931884766 2.0399999618530273 1.590000033378601 2009 January 44.95000076293945 2.0299999713897705 1.5800000429153442 2009 February 46.52000045776367 2.059999942779541 1.6100000143051147 2009 March 49.70000076293945 2.069999933242798 1.6200000047683716 2009 April 51.15999984741211 1.8600000143051147 1.409999966621399 2009 May 60.02000045776367 2.2200000286102295 1.7699999809265137 2009 June 72.23999786376953 2.1700000762939453 1.7200000286102295 2009 July 66.5199966430664 2.140000104904175 1.690000057220459 2009 August 74 2.119999885559082 1.6699999570846558 2009 September 70.22000122070312 2.180000066757202 1.7300000190734863 2009 October 78.25 2.2799999713897705 1.8300000429153442 2009 November 78.11000061035156 2.1500000953674316 1.7000000476837158 2009 December 75.11000061035156 2.049999952316284 1.600000023841858 2010 January 77.62000274658203 2.3299999237060547 1.8799999952316284 2010 February 75.05999755859375 2.390000104904175 1.940000057220459 2010 March 80.2699966430664 2.440000057220459 1.9900000095367432 2010 April 85.29000091552734 2.4100000858306885 1.9600000381469727 2010 May 77.54000091552734 2.4100000858306885 1.9600000381469727 2010 June 75.79000091552734 2.2100000381469727 1.7599999904632568 2010 July 77.18000030517578 2.4800000190734863 2.0299999713897705 2010 August 78.66999816894531 2.5 2.049999952316284 2010 September 79.44999694824219 2.4800000190734863 2.0299999713897705 2010 October 84.41999816894531 2.880000114440918 2.430000066757202 2010 November 86.70999908447266 2.5 2.049999952316284 2010 December 92.79000091552734 2.5799999237060547 2.130000114440918 2011 January 97.95999908

In [6]:


--- total crude oil production and total crude oil export in 2020 
SELECT SUM(Production) AS [total production 2020],
    SUM (Crude_Oil_Export) AS [total export 2020]
FROM crudeoilprice.dbo.CrudeOilProdPrice
WHERE [Year] = 2020

--- percentage of exported oil from produced in 2020
SELECT Month, Production AS [production 2020],
    Crude_Oil_Export AS [crude export 2020],
    (Crude_Oil_Export / Production) * 100  AS [% export per month 2020]
FROM crudeoilprice.dbo.CrudeOilProdPrice
WHERE Year = 2020

(1 row affected)

(12 rows affected)

Total execution time: 00:00:00.026

total production 2020 total export 2020 21.06999969482422 15.670000195503235

Month production 2020 crude export 2020 % export per month 2020 January 2.069999933242798 1.6200000047683716 78.26087231947538 February 2.069999933242798 1.6200000047683716 78.26087231947538 March 2.0399999618530273 1.590000033378601 77.94117956425498 April 2.0399999618530273 1.590000033378601 77.94117956425498 May 1.75 1.2999999523162842 74.28571156093052 June 1.690000057220459 1.2400000095367432 73.37277914511847 July 1.649999976158142 1.2000000476837158 72.72727666807575 August 1.649999976158142 1.2000000476837158 72.72727666807575 September 1.5399999618530273 1.090000033378601 70.77922469991769 October 1.6100000143051147 1.159999966621399 72.04968672761544 November 1.5399999618530273 1.090000033378601 70.77922469991769 December 1.4199999570846558 0.9700000286102295 68.30986323420018

In [7]:
--- Year with the highest crude oil produced
SELECT DISTINCT (Year), SUM(Production)
  OVER (PARTITION BY Year) AS [cumulative total production],
  SUM(Crude_Oil_Export) OVER  (PARTITION BY Year) AS [total exported cruide oil]
 
FROM crudeoilprice.dbo.CrudeOilProdPrice

(15 rows affected)

Total execution time: 00:00:00.037

Year cumulative total production total exported cruide oil 2006 28.59000015258789 23.19000017642975 2007 26.420000076293945 21.020000100135803 2008 25.19000005722046 19.790000081062317 2009 25.330000042915344 19.93000018596649 2010 29.610000371932983 24.210000157356262 2011 28.56999969482422 23.169999957084656 2012 27.820000171661377 22.420000076293945 2013 26.199999809265137 20.799999952316284 2014 26.47000002861023 21.070000052452087 2015 25.519999980926514 20.12000000476837 2016 21.799999833106995 16.40000009536743 2017 22.670000195503235 17.269999861717224 2018 22.980000019073486 17.579999923706055 2019 24.149999856948853 18.75 2020 21.06999969482422 15.670000195503235

The table shown above  is a representation of the cumulative total production of crude oil from from 2006 - 2021. The year with the highest oil production is 2010 with 29.6million barrels. according to the table shown above 2016 produced the lowest crude oil  produced in Nigeria between 2006 and 2020.

In [ ]:
---highest oil price.
SELECT Year, Month, max(Crude_Oil_Price) AS [Maximum Oil Price]
FROM crudeoilprice.dbo.CrudeOilProdPrice
GROUP BY Year, [Month]
ORDER BY max(Crude_Oil_Price) DESC


(180 rows affected)

Total execution time: 00:00:00.057

Year Month Maximum Oil Price 
 2008 June 138.74000549316406 
 2008 July 137.74000549316406 
 2012 March 128 
 2008 May 126.56999969482422 
 2011 April 124.48999786376953 
 2012 April 122.62000274658203 
 2012 February 121.87000274658203 
 2013 February 118.80999755859375 
 2011 May 118.43000030517578 
 2011 July 117.86000061035156 
 2011 June 117.02999877929688 
 2008 April 116.7300033569336 
 2011 March 116.55999755859375 
 2008 August 115.83999633789062 
 2011 September 115.7300033569336 
 2013 January 115.23999786376953 
 2014 June 114.5999984741211 
 2012 December 114.48999786376953 
 2012 September 114.36000061035156 
 2011 November 113.91999816894531 
 2012 January 113.80999755859375 
 2012 August 113.76000213623047 
 2013 September 113.58999633789062 
 2011 October 113.12000274658203 
 2012 May 113.08000183105469 
 2013 March 112.79000091552734 
 2013 December 112.75 
 2013 October 112.29000091552734 
 2011 August 111.98999786376953 
 2014 May 111.9000015258789 
 2011 December 111.45999908447266 
 2013 November 111.13999938964844 
 2012 November 111.05000305175781 
 2014 February 110.83000183105469 
 2014 April 110.41000366210938 
 2014 January 110.19000244140625 
 2013 July 109.77999877929688 
 2014 July 109.62999725341797 
 2014 March 109.47000122070312 
 2012 October 108.91999816894531 
 2013 August 107.83999633789062 
 2011 February 106.56999969482422 
 2013 June 106.05999755859375 
 2013 May 106 
 2013 April 105.55000305175781 
 2012 July 104.62000274658203 
 2008 September 103.81999969482422 
 2008 March 103.7300033569336 
 2014 August 102.33000183105469 
 2014 September 98.2699966430664 
 2008 February 98.1500015258789 
 2012 June 98.05999755859375 
 2011 January 97.95999908447266 
 2007 November 95.05000305175781 
 2008 January 94.26000213623047 
 2007 December 93.4000015258789 
 2010 December 92.79000091552734 
 2010 November 86.70999908447266 
 2010 April 85.29000091552734 
 2010 October 84.41999816894531 
 2007 October 83.86000061035156 
 2014 October 83.5 
 2014 November 80.41999816894531 
 2010 March 80.2699966430664 
 2007 July 79.76000213623047 
 2007 September 79.76000213623047 
 2018 September 79.58999633789062 
 2010 September 79.44999694824219 
 2018 October 79.18000030517578 
 2010 August 78.66999816894531 
 2009 October 78.25 
 2009 November 78.11000061035156 
 2018 May 77.63999938964844 
 2010 January 77.62000274658203 
 2010 May 77.54000091552734 
 2010 July 77.18000030517578 
 2010 June 75.79000091552734 
 2018 June 75.37999725341797 
 2008 October 75.30999755859375 
 2006 August 75.1500015258789 
 2006 July 75.12999725341797 
 2009 December 75.11000061035156 
 2010 February 75.05999755859375 
 2018 July 74.72000122070312 
 2018 March 74.72000122070312 
 2009 August 74 
 2007 August 73.76000213623047 
 2019 May 73.6500015258789 
 2018 August 73.3499984741211 
 2007 June 73.27999877929688 
 2019 April 73.08000183105469 
 2018 April 72.37000274658203 
 2009 June 72.23999786376953 
 2006 April 72.08999633789062 
 2006 May 71.18000030517578 
 2007 April 70.45999908447266 
 2007 May 70.4000015258789 
 2009 September 70.22000122070312 
 2018 January 69.68000030517578 
 2006 June 69.31999969482422 
 2019 December 68.55999755859375 
 2019 March 67.66999816894531 
 2019 June 66.73999786376953 
 2020 January 66.68000030517578 
 2018 February 66.66999816894531 
 2018 November 66.58999633789062 
 2009 July 66.5199966430664 
 2019 July 66.23999786376953 
 2019 September 65.2699966430664 
 2017 December 65.11000061035156 
 2015 May 65.08000183105469 
 2006 March 65 
 2019 February 64.88999938964844 
 2006 December 64.69999694824219 
 2007 March 64.27999877929688 
 2006 January 63.849998474121094 
 2019 November 63.560001373291016 
 2017 November 63.560001373291016 
 2014 December 63.279998779296875 
 2006 September 62.970001220703125 
 2015 June 62.060001373291016 
 2018 December 62 
 2006 February 61.33000183105469 
 2019 August 61.04999923706055 
 2019 January 60.38999938964844 
 2009 May 60.02000045776367 


In June 2008 the price of oil was  $138.7 which appears to be the highest price of oil betwen 2006  and  2020. while April 2020 recorded the lowest prices of oil at $14.27.